src/crystal/system/unix/epoll/event_queue.cr

```diff
@@ -0,0 +1,103 @@
+{% skip_file unless flag?(:linux) || flag?(:solaris) %}
+
+require "./event"
+
+# NOTE: this is a struct because it only wraps a const pointer to a hash
+# allocated in the heap —no need to allocate an object and dereference 2
+# pointers when we can only dereference 1.
+struct Crystal::Epoll::EventQueue
+  class Node
+    property fd : Int32
+    property events : Int32 = 0
+
+    getter readers = PointerLinkedList(Event).new
+    getter writers = PointerLinkedList(Event).new
```

Comment:
Do these need to be protected behind locks in preview_mt, perhaps?

Reply:
The data structure isn't thread-safe. All accesses to `EventQueue`, `EventQueue::Node` and `Event` (when shared) should be protected by a Thread::Mutex in EventLoop.